wgpu-core/src/instance.rs

```diff
@@ -250,6 +250,44 @@ impl<B: GfxBackend> Adapter<B> {
         }
     }
 
+    pub fn get_swap_chain_preferred_format(
+        &self,
+        surface: &mut Surface,
```

Comment:
does this need to be mutable?

Reply:
Not necessarily, but I only found the `get_surface_mut` function for getting the gfx-hal surface of a wgpu surface. If there's an immutable alternative, I'd be glad to use it.

Reply:
that function doesn't force you to accept `&mut` though, you can still have a `&` argument

Reply:
Not sure if I'm missing something, but get_surface_mut only appears to take &mut as input. How would I take an immutable reference and pass it as a mutable reference if the type is not copyable?

Reply:
oh, sorry, I see now :)